cmd/git-sync/signal.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+var signalMap = map[string]int{
+	"SIGHUP":    1,
+	"SIGINT":    2,
+	"SIGQUIT":   3,
+	"SIGILL":    4,
+	"SIGTRAP":   5,
+	"SIGABRT":   6,
+	"SIGBUS":    7,
+	"SIGFPE":    8,
+	"SIGKILL":   9,
+	"SIGUSR1":   10,
+	"SIGSEGV":   11,
+	"SIGUSR2":   12,
+	"SIGPIPE":   13,
+	"SIGALRM":   14,
+	"SIGTERM":   15,
+	"SIGSTKFLT": 16,
+	"SIGCHLD":   17,
+	"SIGCONT":   18,
+	"SIGSTOP":   19,
+	"SIGTSTP":   20,
+	"SIGTTIN":   21,
+	"SIGTTOU":   22,
+	"SIGURG":    23,
+	"SIGXCPU":   24,
+	"SIGXFSZ":   25,
+	"SIGVTALRM": 26,
+	"SIGPROF":   27,
+	"SIGWINCH":  28,
+	"SIGIO":     29,
+	"SIGPWR":    30,
+	"SIGSYS":    31,
+}
+
+func ConvertSignal(flProcSignal string) (syscall.Signal, error) {
+	sig, err := strconv.ParseInt(flProcSignal, 10, 32)
+	if err == nil {
+		return syscall.Signal(sig), nil
+	} else {
+		if sig, ok := signalMap[flProcSignal]; ok {
+			return syscall.Signal(sig), nil
+		}
+	}
+	return syscall.Signal(1), nil
+}
+
+func SignalProcs(flProcName string, sig syscall.Signal) error {
+	pids, err := getPids()
+	if err != nil {
+		return err
+	}
+	for _, pid := range pids {
+		if getName(pid) == flProcName {
+			err := syscall.Kill(int(pid), sig)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func getHostProc() string {
+	procDir, found := os.LookupEnv("HOST_PROC")
```

Comment:
This is not a standard that I know of?  If not, it should be be prefixed with `GIT_SYNC_`, eg GIT_SYNC_PROCFS_PATH

Reply:
I guess I saw it in Telegraf, and thought it made enough sense that it was a standard.  Agree that it's not after future research, will modify.